### PR TITLE
qemu: Fix sphinx and sphinx-rtd-theme deps for cross

### PR DESCRIPTION
**Pull request by dramforever**
###### Things done

- Built on platform(s)
  - [X] x86_64-linux (cross to riscv64-linux)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/virtualization/qemu/default.nix [hunk 1]
```diff
@@ -50,7 +50,7 @@ stdenv.mkDerivation rec {
 
   depsBuildBuild = [ buildPackages.stdenv.cc ];
 
-  nativeBuildInputs = [ makeWrapper removeReferencesTo pkg-config flex bison meson ninja perl python3 python3.pkgs.sphinx python3.pkgs.sphinx-rtd-theme ]
+  nativeBuildInputs = [ makeWrapper removeReferencesTo pkg-config flex bison meson ninja perl python3 buildPackages.python3.pkgs.sphinx buildPackages.python3.pkgs.sphinx-rtd-theme ]
```
Comment (alyssais) [inline pkgs/applications/virtualization/qemu/default.nix:53]: It seems strange that we'd need this.  Do you know why splicing doesn't take care of it for us?

Comment (dramforever) [reply pkgs/applications/virtualization/qemu/default.nix:53]: Apparently only the argument `python3` is spliced, not the 'nested' derivations

**Comment by nikstur** [comment]
I think this can be closed since the underlying issue was fixed in https://github.com/NixOS/nixpkgs/commit/f70071e41da960a1d68d3cef0cb1e2e4d0d93fbf

**Comment by dramforever** [comment]
Yes sorry, this is an outdated PR I forgot to close.

**Comment by nikstur** [comment]
No worries. Thank you for closing it!

**Pull request closed without merging**